src/test/kotlin/no/nav/familie/ef/sak/ekstern/bisys/BisysBarnetilsynServiceTest.kt

```diff
@@ -298,35 +297,41 @@ internal class BisysBarnetilsynServiceTest {
     }
 
     @Test
-    fun `en infotrygdperiode og en andelshistorikk etter fraOgMedDato i oppslag, forvent sammenslåtte perioder`() {
-        mockTilkjentYtelse()
+    fun `en infotrygdperiode som strekker seg forbi ef-perioder, forvent perioder ikke varer lenger enn ef-perioder`() {
+        val fraOgMed = YearMonth.now().atDay(1).minusMonths(1)
+        mockTilkjentYtelse(fraOgMed)
+        val tilOgMedDato = YearMonth.now().atEndOfMonth().plusMonths(2)
         val andelhistorikkDto =
             lagAndelHistorikkDto(
-                fraOgMed = LocalDate.MIN.plusDays(1),
-                tilOgMed = LocalDate.now().plusMonths(2),
-                behandlingBarn = behandlingBarn
+                fraOgMed = fraOgMed,
+                tilOgMed = tilOgMedDato,
+                behandlingBarn = behandlingBarn,
+                beløp = 10
             )
         every {
             infotrygdService.hentSammenslåtteBarnetilsynPerioderFraReplika(any())
         } returns listOf(
             lagInfotrygdPeriode(
                 vedtakId = 1,
                 stønadFom = LocalDate.MIN,
-                stønadTom = LocalDate.now().plusMonths(1),
+                stønadTom = LocalDate.now().plusMonths(6),
                 beløp = 10
             )
         )
         every {
             andelsHistorikkService.hentHistorikk(any(), any())
         } returns listOf(andelhistorikkDto)
 
-        val fomDato = LocalDate.now()
-        assertThat(
-            barnetilsynBisysService.hentBarnetilsynperioderFraEfOgInfotrygd(
-                personident,
-                fomDato
-            ).barnetilsynBisysPerioder
-        ).hasSize(2)
+        val perioder = barnetilsynBisysService.hentBarnetilsynperioderFraEfOgInfotrygd(
+            personident,
+            LocalDate.MIN
+        ).barnetilsynBisysPerioder
+        assertThat(perioder).hasSize(2)
+        // Sjekk at overgang fra infotrygperiode til ny løsning blir riktig
+        assertThat(perioder.last().periode.fom).isEqualTo(fraOgMed)
+        assertThat(perioder.first().periode.tom).isEqualTo(fraOgMed.minusDays(1))
+        // selv om infotrygdperioden varer lenger har ny løsning tatt over og sier vi er ferdige her:
+        assertThat(perioder.last().periode.tom).isEqualTo(tilOgMedDato)
```

Comment:
Er kommentarene nødvendige? 
1. Tenker testnavnet burde være tilstrekkelig beskrivende av hva det er man tester
2. Hvis man trenger kommentarer, kanskje det burde være en del av koden, https://joel-costigliola.github.io/assertj/assertj-core-features-highlight.html#describe-assertion men føler ikke helt det er nødvendig her
3. Jeg tror generellt tester blir enklere hvis man grupperer de for lesbarhet, og tar bort unødvendig kode til hjelpemetoder
```kotlin
@Test
    fun `efperioder skal avkorte infotrygdperioder hvis barnen er ulike`() {
        @Test
    fun `efperioder skal avkorte infotrygdperioder`() {
        val infotrygdFra = LocalDate.of(2021, 1, 1)
        val infotrygdTil = LocalDate.of(2022, 1, 31)

        val fom = infotrygdFra.plusMonths(6)
        val tom = LocalDate.of(2023, 1, 31)

        mockTilkjentYtelse(fom)
        mockInfotrygd(lagInfotrygdPeriode(stønadFom = infotrygdFra, stønadTom = infotrygdTil))
        mockHentHistorikkAndel(
            lagAndelHistorikkDto(
                fraOgMed = fom,
                tilOgMed = tom,
                behandlingBarn = behandlingBarn,
            )
        )

        val perioder = barnetilsynBisysService.hentBarnetilsynperioderFraEfOgInfotrygd(personident, infotrygdFra)
            .barnetilsynBisysPerioder.map { it.periode }

        assertThat(perioder).hasSize(2)
        assertThat(perioder.first().fom).isEqualTo(infotrygdFra)
        assertThat(perioder.first().tom).isEqualTo(fom.minusDays(1))
        assertThat(perioder.last().fom).isEqualTo(fom)
        assertThat(perioder.last().tom).isEqualTo(tom)
    }
    }
```

Så kan man vurdere bruk av YearMonth og/eller sette datoer annerledes, eks
```
val infotrygdFra = LocalDate.of(2021, 1, 1)
val infotrygdTil = LocalDate.of(2023, 1, 31)

val fom = infotrygdFra.plusMonths(6)
val tom = infotrygdTil.plusYears(1)
```

Eventuellt dette koblet til asserts
```
val periodeFraInfotrygd = perioder.first()
val periodeFraEf = perioder.last()
```

Men for att denne testen skal være gyldig, så er den ju avhengig av att man setter `behandlingBarn = behandlingBarn,` som gjør at barnen på periodene fra infotrygd og EF er ulike. 
Hvis ikke så slås de sammen

Reply:
Nesten så man burde brukt dette for ennå mer lesbarhet i våre tester
```
val JAN = YearMonth.of(2021, 1)
val JAN_2021 = YearMonth.of(2021, 1) // der årstall er nødvendig
```

Reply:
Da trakk jeg ut noen hjelpemetoder og erstattet duplikat kode eller i fila. 
Ble flere endringer enn planlagt, men tror testene totalt sett er lettere å lese nå.